File: DTOs/Sources/RideServiceDTOs/GetSpecialRideDetailDTO.swift
import Foundation 

public struct GetSpecialRideDetailDTO: Codable {
    public var id: UUID?
    public var driverName: String
    public var driverID: UUID
    public var isSelfDriver: Bool
    public var name: String
    public var description: String?
    public var vehicleDescription: String?
    public var starts: Date
    public var ends: Date
    public var startLatitude: Float
    public var startLongitude: Float
    public var destinationLatitude: Float
    public var destinationLongitude: Float
    public var emptySeats: UInt8
    public var riders: [GetRiderDTO]
    
    public init(id: UUID? = nil, driverName: String, driverID: UUID, isSelfDriver: Bool, name: String, description: String? = nil, vehicleDescription: String? = nil, starts: Date, ends: Date, startLatitude: Float, startLongitude: Float, destinationLatitude: Float, destinationLongitude: Float, emptySeats: UInt8, riders: [GetRiderDTO]) {
        self.id = id
        self.driverName = driverName
        self.driverID = driverID
        self.isSelfDriver = isSelfDriver
        self.name = name
        self.description = description
        self.vehicleDescription = vehicleDescription
        self.starts = starts
        self.ends = ends
        self.startLatitude = startLatitude
        self.startLongitude = startLongitude
        self.destinationLatitude = destinationLatitude
        self.destinationLongitude = destinationLongitude
        self.emptySeats = emptySeats
        self.riders = riders
    }
    
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/RideServiceDTOs/GetSpecialRideDetailDTO.kt
package net.ipv64.kivop.dtos.RideServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

//import Foundation 
data class GetSpecialRideDetailDTO (
    var id : UUID,
    var driverName : String,
    var driverID : UUID,
    var isSelfDriver : Boolean,
    var name : String,
    var description : String?,
    var vehicleDescription : String?,
    var starts : LocalDateTime,
    var ends : LocalDateTime,
    var startLatitude : Float,
    var startLongitude : Float,
    var destinationLatitude : Float,
    var destinationLongitude : Float,
    var emptySeats : UByte,
    var riders : List<GetRiderDTO>,
)
